File: Cpp/55_Jump_Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_index = 0;
        for(int i=0; i<nums.size(); ++i) {
            if(max_index < i) break;
            max_index = max(max_index, i + nums[i]); 
        }
        
        return max_index >= nums.size() - 1;
    }
};

File: Kotlin/55_Jump_Game.kt
class Solution {
    fun canJump(nums: IntArray): Boolean {
        var maxIndex = 0;
        for(i in 0 until nums.size){
            if(maxIndex < i) break
            maxIndex = maxOf(maxIndex, i + nums[i])
        }
        
        return maxIndex >= nums.size - 1
    }
}